compiler/src/iree/compiler/Codegen/Dialect/GPU/IR/IREEGPUAttrs.cpp

```diff
@@ -1054,75 +888,39 @@ MMAScheduleAttr::getContractionLayout(vector::ContractionOp contractOp) const {
   SmallVector<int64_t> aMDims = opInfo.lhsMDims;
   SmallVector<int64_t> aBatchSizes(aRank, 1);
   SmallVector<int64_t> aSubgroupSizes(aRank, 1);
-  SmallVector<int64_t> aSubgroupOrder = getIdentityPerm(aRank);
-  SmallVector<int64_t> aBatchOrder = getIdentityPerm(aRank);
+  SmallVector<int64_t> aSubgroupStrides(aRank, 1);
   for (auto [i, dim] : llvm::enumerate(aMDims)) {
     aBatchSizes[dim] = batchMSizes[i];
     aSubgroupSizes[dim] = subgroupMBasis[i];
-    int64_t j = i + batchCount;
-    aSubgroupOrder[dim] = j;
-    aBatchOrder[dim] = j >= afk ? j + 1 : j;
+    aSubgroupStrides[dim] = subgroupMStrides[i];
   }
-  aSubgroupOrder[afk] = aRank - 1;
   aBatchSizes[afk] = bounds[opInfo.getKDims().back()] / intrinsicK;
 
-  SmallVector<bool> aActiveSubgroups(subgroupBasis.size(), false);
-  for (auto mDim : mDims) {
-    aActiveSubgroups[mDim] = true;
-  }
-  aActiveSubgroups.back() = true;
-  if (batchCount == 1) {
-    aActiveSubgroups[0] = true;
-  }
-
-  auto aLayout = permuteAndCreateNestedLayout(
-      context, aRank, afm, afk,
-      /*subgroupCount=*/aSubgroupSizes,
-      /*subgroupOrder=*/aSubgroupOrder,
-      /*batchCount=*/aBatchSizes, aCounts, aOrders,
-      /*dataDuplicate=*/mmaAttr.getADataDuplicate(), subgroupBasis,
-      aActiveSubgroups);
+  auto aLayout = createNestedLayout(context, aRank, afm, afk,
+                                    /*subgroupCount=*/aSubgroupSizes,
+                                    /*subgroupStrides=*/aSubgroupStrides,
+                                    /*batchCount=*/aBatchSizes,
+                                    mmaAttr.getASingleSubgroupLayout());
   LLVM_DEBUG({ llvm::errs() << "A layout: " << aLayout << "\n"; });
 
-  // B matrix layout
-  MMAAttr::SingleSubgroupLayout bCounts =
-      mmaAttr.getBSingleSubgroupLayoutCount();
-  MMAAttr::SingleSubgroupLayout bOrders =
-      mmaAttr.getBSingleSubgroupLayoutOrder();
-
   int64_t bRank = opInfo.getBRank();
 
   SmallVector<int64_t> bNDims = opInfo.rhsNDims;
   SmallVector<int64_t> bBatchSizes(bRank, 1);
   SmallVector<int64_t> bSubgroupSizes(bRank, 1);
-  SmallVector<int64_t> bSubgroupOrder = getIdentityPerm(bRank);
-  SmallVector<int64_t> bBatchOrder = getIdentityPerm(bRank);
+  SmallVector<int64_t> bSubgroupStrides(bRank, 1);
```

Comment:
If we have subgroupStride = 1 on the K-dims, Does this mean for every time we move in the K-dim, we are moving by 1 subgroup?

Reply:
You are right, i messed up here... this should be 0